client/src/features/Home/components/CardContainer/CardContainer.js

```diff
@@ -80,60 +40,92 @@ const CardContainerSwiperContainer = styled('div')({
   },
 });
 
-const CardContainer = () => (
-  <Container>
-    <CardContainerHeaderContainer>
-      <Typography variant="h2" sx={{ marginBottom: '0px' }}>
-        Popular tours
-      </Typography>
-      <Link to="/catalogue">
-        <Typography>View All</Typography>
-      </Link>
-    </CardContainerHeaderContainer>
-    <CardContainerSwiperContainer>
-      <SliderButton id="previous" position={{ top: '44%', left: '-30px', rotate: '180deg' }} />
-      <SliderButton id="next" position={{ top: '44%', right: '-30px' }} />
-      <Swiper
-        breakpoints={{
-          265: {
-            width: 265,
-            slidesPerView: 1,
-          },
-          768: {
-            width: 560,
-            slidesPerView: 2,
-          },
-          1024: {
-            width: 855,
-            slidesPerView: 3,
-          },
-          1360: {
-            width: 1150,
-            slidesPerView: 4,
-          },
-        }}
-        spaceBetween={30}
-        slidesPerGroup={1}
-        loop
-        loopFillGroupWithBlank
-        pagination={{
-          clickable: true,
-        }}
-        navigation={{
-          nextEl: '#next',
-          prevEl: '#previous',
-        }}
-        modules={[Navigation]}
-        className="mySwiper"
-      >
-        {cards.map(({ name, region, id }) => (
-          <SwiperSlide key={id}>
-            <CardItem name={name} region={region} id={id} />
-          </SwiperSlide>
-        ))}
-      </Swiper>
-    </CardContainerSwiperContainer>
-  </Container>
-);
+const CardContainer = () => {
+
+  const dispatch = useDispatch();
+  const products = useSelector((state) => state.catalogue.products, shallowEqual);
+  const isLoading = useSelector((state) => state.catalogue.isLoading);
+  const [currentPage] = useState(1);
+  const countriesPerPage = 10;
+  const lastItemIndex = currentPage * countriesPerPage;
+  const firstItemIndex = lastItemIndex - countriesPerPage;
+  const currentItems = products.slice(firstItemIndex, lastItemIndex);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (products.length <= 0) {
+      dispatch(setIsLoading(true));
+      dispatch(getProducts());
+      dispatch(setIsLoading(false));
+    }
+  }, []);
```

Comment:
i dont think u need this effect here also as it makes sense only in catalogue page,  in our home page your section is the third section. user will never see the loading state.

Reply:
done